app/helpers/hosts_and_hostgroups_helper.rb

```diff
@@ -103,8 +103,8 @@ def realm_field(f, can_override = false, override = false)
   def interesting_klasses(obj)
     classes    = obj.all_puppetclasses
     classes_ids = classes.reorder('').pluck('puppetclasses.id')
-    smart_vars = LookupKey.reorder('').where(:puppetclass_id => classes_ids).uniq.pluck(:puppetclass_id)
-    class_vars = LookupKey.reorder('').joins(:environment_classes).where(:environment_classes => { :puppetclass_id => classes_ids }).uniq.pluck('environment_classes.puppetclass_id')
+    smart_vars = VariableLookupKey.reorder('').where(:puppetclass_id => classes_ids).uniq.pluck(:puppetclass_id)
+    class_vars = PuppetclassLookupKey.reorder('').joins(:environment_classes).where(:environment_classes => { :puppetclass_id => classes_ids }).uniq.pluck('environment_classes.puppetclass_id')
     klasses    = (smart_vars + class_vars).uniq
```

Comment:
it could have an effect on large data set, if you could give it a try, maybe with the context of #2664